### PR TITLE
applied during french translation doc/en@14e5e29fd12657263368b305369d…

### PR DESCRIPTION
**Pull request by eugene-augier**
See: https://github.com/php/doc-fr/pull/211

### CODE REVIEW

**Comment by cmb69** [comment]
Do we have any style guides which prefer one comment style over the other? Otherwise, it's not unlikely that we see a commit in the future which changes the style again.

cc @kamil-tekiela

**Review by kamil-tekiela — CHANGES_REQUESTED** [review]
I am not sure what comment style is mandated by PHP Manual, but in general, we use single-line comment syntax for single-line comments and multi-line syntax for multi-line comments. 

These are good changes, but that one example needs more changes. I seem to have missed it in my redesign.

**Comment by cmb69** [comment]
> I am not sure what comment style is mandated by PHP Manual, but in general, we use single-line comment syntax for single-line comments and multi-line syntax for multi-line comments.

The [style guidelines](http://doc.php.net/tutorial/style.php) are terse, and just refer to [PEAR CS](https://pear.php.net/manual/en/standards.comments.php) which explicitly allows both styles. We might be better off to document our preferred style in the style guidelines.

**Comment by eugene-augier** [comment]
Generally speaking, single-line comments used at the layer of PHP frameworks are declared with `//`, that's why I proposed these changes. 

Of course I understand that the php documentation is an independent project from the upper software layers.

**Comment by cmb69** [comment]
I'm not against these changes, but I'm concerned that a follow-up PR/commit might change back to some other style (and such comments are a burden for translators). Therefore I think we need to make a decision, and to document it. This should probably be discussed on the phpdoc mailing list.

@Girgias, do we already have scripts to check code examples?

**Comment by eugene-augier** [comment]
I completely understand your point.

File: reference/mysqli/quickstart.xml [hunk 1]
```diff
@@ -1472,12 +1473,13 @@ array(1) {
     <programlisting role="php">
 <![CDATA[
 <?php
+
+mysqli_report(MYSQLI_REPORT_ERROR | MYSQLI_REPORT_STRICT);
```
[the remark this answers is not in the document]
Comment (kamil-tekiela) [reply reference/mysqli/quickstart.xml:1477]: No, it's better if you don't. BTW if you want to use try-catch, use it only for the line you expect to throw, not the whole code.

**Review by kamil-tekiela — APPROVED** [review]

**Comment by Girgias** [comment]
> I'm not against these changes, but I'm concerned that a follow-up PR/commit might change back to some other style (and such comments are a burden for translators). Therefore I think we need to make a decision, and to document it. This should probably be discussed on the phpdoc mailing list.
> 
> @Girgias, do we already have scripts to check code examples?

We have a linter check if code segment compile, but it doesn't check the style, (nor the output). But that would be a good script to have

**Comment by cmb69** [comment]
> But that would be a good script to have

I very much agree. But that would likely bring us back to the "we must not use existing apps, to not endorse them". :( And writing our own CS checker appears to be overkill.

**Comment by eugene-augier** [comment]
> > I'm not against these changes, but I'm concerned that a follow-up PR/commit might change back to some other style (and such comments are a burden for translators). Therefore I think we need to make a decision, and to document it. This should probably be discussed on the phpdoc mailing list.
> > @Girgias, do we already have scripts to check code examples?
> 
> We have a linter check if code segment compile, but it doesn't check the style, (nor the output). But that would be a good script to have

I am open to work on this project.
We could check that the php code is syntactically valid by the way, that would avoid some errors.
```php
$url = "path"/;
```

**Comment by Girgias** [comment]
> > > I'm not against these changes, but I'm concerned that a follow-up PR/commit might change back to some other style (and such comments are a burden for translators). Therefore I think we need to make a decision, and to document it. This should probably be discussed on the phpdoc mailing list.
> > > @Girgias, do we already have scripts to check code examples?
> > 
> > 
> > We have a linter check if code segment compile, but it doesn't check the style, (nor the output). But that would be a good script to have
> 
> I am open to work on this project. We could check that the php code is syntactically valid by the way, that would avoid some errors.
> 
> ```
> $url = "path"/;
> ```

Great, the script is located in doc-base and the current one we have is: https://github.com/php/doc-base/blob/master/scripts/qa/check-examples.php which checks for syntactically valid code (and no errors), which is a good start, but some code examples fail on purpose and so currently the whole page need to be silenced :|

**Comment by eugene-augier** [comment]
Hi @cmb69 

After some private discussions with Girgias, I would like to propose some changes on `php/doc-base`.
When you have the time and desire, this is the one I mentioned above.

Thanks!

**Comment by kamil-tekiela** [comment]
Can we merge this?

**Comment by cmb69** [comment]
I still don't think we should make cosmetic changes without having some kind of style guide which justifies these changes. Thus, I'm going to commit only the relevant parts of the PR.

Thanks all!

**Pull request closed without merging**